libnopegl/src/params.c

```diff
@@ -164,9 +164,9 @@ const struct param_specs ngli_params_specs[] = {
     },
 };
 
-static const char *get_param_type_name(int param_type)
+static const char *get_param_type_name(enum param_type param_type)
 {
-    if (param_type < 0 || param_type >= NGLI_ARRAY_NB(ngli_params_specs))
+    if (param_type >= NGLI_ARRAY_NB(ngli_params_specs))
```

Comment:
I'm sure we can still call the function with a negative number and make it crash. Maybe we have a warning if we're not passing the enum itself, but in that case the upper boundary is unnecessary as well? We could probably replace this if by an assert anyway (it doesn't seem to be user controlled) or remove the condition entirely.